core/src/autogluon/core/task/base/base_task.py

```diff
@@ -63,6 +63,8 @@ def run_fit(cls, train_fn, search_strategy, scheduler_options,
         args.final_fit = True
         if hasattr(args, 'epochs') and hasattr(args, 'final_fit_epochs'):
             args.epochs = args.final_fit_epochs
+        train_fn.args.update({'final_fit':True})
+        train_fn.kwvars.update({'final_fit':True})
         scheduler_final = create_scheduler(train_fn, search_strategy, scheduler_options)
```

Comment:
Are you sure this actually fixes the problem? scheduler_final wouldn't have searched that space so it would have to train it from scratch still.

Reply:
I tried locally and it fixes the problem. When providing the final_fit, gluoncv won't actually fit anything. It will reload the best models trained so far instead. I know this is confusing :(.